client/my-sites/stats/feedback/index.tsx

```diff
@@ -1,42 +1,108 @@
+}
+
+interface FeedbackPanelProps {
+	clickHandler: ( action: string ) => void;
+	isOpen: boolean;
+}
+
+function FeedbackContent( { clickHandler }: FeedbackProps ) {
 	const translate = useTranslate();
-	const [ isOpen, setIsOpen ] = useState( false );
 
 	const ctaText = translate( 'How do you rate your overall experience with Jetpack Stats?' );
 	const primaryButtonText = translate( 'Love it? Leave a review' );
 	const secondaryButtonText = translate( 'Not a fan? Help us improve' );
 
-	const handleClickWriteReview = () => {};
+	const handleLeaveReview = () => {
+		clickHandler( FEEDBACK_ACTION_LEAVE_REVIEW );
+	};
 
-	const handleClickSendFeedback = () => {
-		setIsOpen( true );
+	const handleSendFeedback = () => {
+		clickHandler( FEEDBACK_ACTION_SEND_FEEDBACK );
 	};
 
 	return (
-		<div className="stats-feedback-container">
-			<div className="stats-feedback-card">
-				<div className="stats-feedback-card__cta">{ ctaText }</div>
-				<div className="stats-feedback-card__actions">
-					<Button variant="secondary" onClick={ handleClickWriteReview }>
-						<span className="stats-feedback-card__emoji">😍</span>
-						{ primaryButtonText }
-					</Button>
-					<Button variant="secondary" onClick={ handleClickSendFeedback }>
-						<span className="stats-feedback-card__emoji">😠</span>
-						{ secondaryButtonText }
-					</Button>
-				</div>
-				<FeedbackModal isOpen={ isOpen } onClose={ () => setIsOpen( false ) } />
+		<div className="stats-feedback-content">
+			<div className="stats-feedback-content__cta">{ ctaText }</div>
+			<div className="stats-feedback-content__actions">
+				<Button variant="secondary" onClick={ handleLeaveReview }>
+					<span className="stats-feedback-content__emoji">😍</span>
+					{ primaryButtonText }
+				</Button>
+				<Button variant="secondary" onClick={ handleSendFeedback }>
+					<span className="stats-feedback-content__emoji">😠</span>
+					{ secondaryButtonText }
+				</Button>
 			</div>
 		</div>
 	);
 }
 
-export default StatsFeedbackCard;
+function FeedbackPanel( { isOpen, clickHandler }: FeedbackPanelProps ) {
+	const translate = useTranslate();
+
+	const handleCloseButtonClicked = () => {
+		clickHandler( FEEDBACK_ACTION_DISMISS_FLOATING_PANEL );
+	};
+
+	if ( ! isOpen ) {
+		return null;
+	}
+
+	return (
+		<div className="stats-feedback-panel">
+			<Button
+				className="stats-feedback-panel__close-button"
+				onClick={ handleCloseButtonClicked }
+				icon={ close }
+				label={ translate( 'Close' ) }
+			/>
+			<FeedbackContent clickHandler={ clickHandler } />
+		</div>
+	);
+}
+
+function FeedbackCard( { clickHandler }: FeedbackProps ) {
+	return (
+		<div className="stats-feedback-card">
+			<FeedbackContent clickHandler={ clickHandler } />
+		</div>
+	);
+}
+
+function StatsFeedbackController() {
+	const [ isOpen, setIsOpen ] = useState( false );
+	const [ isFloatingPanelOpen, setIsFloatingPanelOpen ] = useState( true );
+
+	const handleButtonClick = ( action: string ) => {
```

Comment:
I'd like to comment again about the pattern here because I don't see much benefit with a general event handler. 

The downsides include,

- All the logic is mix together in a single function, so it grows as you add something more. When the logic in a function could clearly be separated, then we should do so. The pattern is against quite a few principles, like Separation of Concerns, open/close etc.
- The caller has to pass in appropriate parameter to be able to use the function correctly, which makes it harder to use.
- It is a pattern for event driven design; however, imho, they have to do that when the system needs to support customized events, like what the reducer is doing in Redux, which is not applicable here.

Plus if we have to write it in this way, we should probably leverage type checks for action, for example, action: 'action1' | 'action2'.

Plus the function is passed down through multiple levels of components, which can lead to prop drilling issues as the component tree grows. We could probably leverage global state management (probably not the best idea since we are ditching Redux) or custom hooks to share the stats and the functions.

What are your thoughts @Automattic/red?

Reply:
Just to clarify, you'd prefer to see three separate handlers here, right?

I had a peek at the original feedback [here](https://github.com/Automattic/wp-calypso/pull/94219#discussion_r1746199633). I'll take a crack at this in a follow-up PR.

Reply:
> Just to clarify, you'd prefer to see three separate handlers here, right?

Yes exactly. That is the biggest thing in my mind. Thanks @a8ck3n !

Reply:
@kangzj — I've addressed this feedback in the PR linked below. I think it looks pretty clean with those changes in place. Thanks!

https://github.com/Automattic/wp-calypso/pull/94983